src/ui/shield.tsx

```diff
@@ -22,8 +23,11 @@ export default function ShieldedApp() {
     )*/
     let netErr = ""
     if(error && error.networkError) netErr = error.networkError.message + " - "
-    //if (error && !error.message.includes('Not Authorised')) return <h1>{netErr}Error<span className="active">.</span></h1>    
+    if (error && !error.message.includes('Not Authorised')) return <h1>{netErr}Error<span className="active">.</span></h1>    
     let name = data && data.whoami.name
-    return <App user={name}/>
+    let email = data && data.whoami.email
+    let age = data && data.whoami.age
+    const user: User = {name,email,age}
+    return <App user={user}/>
```

Comment:
You can write it as this:
```suggestion
    return <App user={data.whoami}/>
```
You can add the type information to the query response:
`const { loading, error, data } = useQuery(WHO_Q)`
to:
`const { loading, error, data } = useQuery<{  whoami: User }>(WHO_Q)`